src/query/sql/src/planner/optimizer/property/selectivity.rs

```diff
@@ -378,3 +381,58 @@ fn update_statistic(
     }
     Ok(())
 }
+
+fn evaluate_equal_by_histogram(const_datum: &Datum, col_stat: &ColumnStat) -> Result<f64> {
+    let hist = col_stat.histogram.as_ref().unwrap();
+    let min = col_stat.min.to_double()?;
+    let max = col_stat.max.to_double()?;
+    let mut range = 0.0;
+    for (idx, bucket) in hist.buckets_iter().enumerate() {
+        if idx == 0 {
+            continue;
+        }
+        let bucket_min = hist.buckets[idx - 1].upper_bound().to_double()?;
+        let bucket_max = bucket.upper_bound().to_double()?;
+        // If the bucket max is less than min, skip it
+        if bucket_max < min {
```

Comment:
Since `bucket_max`(upper_bound) is derived from `col_stat.min` and col_stat.max, so I think `bucket_max < min` will always be false.